src/proxy/routes/index.js

```diff
@@ -5,13 +5,55 @@ const proxy = require('express-http-proxy');
 const router = express.Router();
 const chain = require('../chain');
 
+/**
+ * Check whether an HTTP request is a Git request. It is a Git request if it
+ * matches one of the following:
+ * - the URL path is a well-known Git endpoint
+ * - the User-Agent starts with "git/"
+ * - the Accept header starts with "application/x-git-" (for specific paths)
+ * @param {express.Request} req Raw HTTP request
+ * @return {boolean} Whether the request is a Git request
+ */
+const validGitRequest = (req) => {
```

Comment:
If specific to GitHub, should we name it accordingly?

Reply:
Good point. I think having a different "URL path sanitizer" function which is specific to GitHub is probably warranted. Other than parsing the path, this is generic to git.